x-pack/legacy/plugins/siem/public/containers/case/api.test.tsx

```diff
@@ -0,0 +1,402 @@
+ */
+
+import { KibanaServices } from '../../lib/kibana';
+import {
+  deleteCases,
+  getActionLicense,
+  getCase,
+  getCases,
+  getCasesStatus,
+  getCaseUserActions,
+  getReporters,
+  getTags,
+  patchCase,
+  patchCasesStatus,
+  patchComment,
+  postCase,
+  postComment,
+  pushCase,
+  pushToService,
+} from './api';
+import {
+  actionLicenses,
+  allCases,
+  basicCase,
+  allCasesSnake,
+  basicCaseSnake,
+  actionTypeExecutorResult,
+  pushedCaseSnake,
+  casesStatus,
+  casesSnake,
+  cases,
+  caseUserActions,
+  pushedCase,
+  pushSnake,
+  respReporters,
+  serviceConnector,
+  casePushParams,
+  tags,
+  caseUserActionsSnake,
+  casesStatusSnake,
+} from './mock';
+import { CASES_URL } from './constants';
+import { DEFAULT_FILTER_OPTIONS, DEFAULT_QUERY_PARAMS } from './use_get_cases';
+
+const abortCtrl = new AbortController();
+const mockKibanaServices = KibanaServices.get as jest.Mock;
+jest.mock('../../lib/kibana');
+
+const fetchMock = jest.fn();
+mockKibanaServices.mockReturnValue({ http: { fetch: fetchMock } });
+
+describe('Case Configuration API', () => {
+  describe('deleteCases', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue('true');
+    });
+    const data = ['1', '2'];
+
+    test('check url, method, signal', async () => {
+      await deleteCases(data, abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}`, {
+        method: 'DELETE',
+        query: { ids: JSON.stringify(data) },
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await deleteCases(data, abortCtrl.signal);
+      expect(resp).toEqual('true');
+    });
+  });
+  describe('getActionLicense', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(actionLicenses);
+    });
+    test('check url, method, signal', async () => {
+      await getActionLicense(abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`/api/action/types`, {
+        method: 'GET',
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await getActionLicense(abortCtrl.signal);
+      expect(resp).toEqual(actionLicenses);
+    });
+  });
+  describe('getCase', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(basicCaseSnake);
+    });
+    const data = basicCase.id;
+
+    test('check url, method, signal', async () => {
+      await getCase(data, true, abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}/${basicCase.id}`, {
+        method: 'GET',
+        query: { includeComments: true },
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await getCase(data, true, abortCtrl.signal);
+      expect(resp).toEqual(basicCase);
+    });
+  });
+  describe('getCases', () => {
```

Comment:
I think we should create a test that tests if the reporters, tags, etc are being formatted correctly. Spefically these lines of code:

```
reporters: filterOptions.reporters.map(r => r.username ?? '').filter(r => r !== ''),
    tags: filterOptions.tags,
    ...(filterOptions.status !== '' ? { status: filterOptions.status } : {}),
    ...(filterOptions.search.length > 0 ? { search: filterOptions.search } : {}),
    ...queryParams,
```